ocs_ci/utility/perf_dash/dashboard_api.py

```diff
@@ -0,0 +1,510 @@
+            topology (str): The topology (e.g. 3-AZ)
+            test (str): The test name (e.g. SmallFiles)
+
+        Returns:
+             dict : dictionary of all test samples as :
+                    {sample: {es_link, log_file},}
+
+        """
+
+        def value_verify(value, msg):
+            if value is None:
+                log.error(f"{msg} does not exist in the DB!")
+                return False
+            else:
+                return True
+
+        ver_id = self.get_version_id(version=version)
+        build_id = self.get_build_id(version=version, build=build)
+        platform_id = self.get_platform_id(platform=platform)
+        topology_id = self.get_topology_id(topology=topology)
+        test_id = self.get_test_id(test=test)
+        if not (
+            value_verify(ver_id, f"Version : {version}")
+            and value_verify(build_id, f"Build : {version}-{build}")
+            and value_verify(platform_id, f"Platform : {platform}")
+            and value_verify(topology_id, f"Topology : {topology}")
+            and value_verify(test_id, f"Test : {test}")
+        ):
+            return None
+        results = {}
+        query = (
+            f"SELECT sample,es_link,log_file FROM results WHERE version = {ver_id} "
+            f"AND build = {build_id} AND platform = {platform_id} AND "
+            f"az_topology = {topology_id} AND test_name = {test_id} ;"
+        )
+        if self._run_query(query=query):
+            for (sample, eslink, logfile) in self.cursor:
+                log.debug(f"{sample}, {eslink}, {logfile}")
+                results[sample] = {
+                    "eslink": eslink.rstrip("\r\n"),
+                    "log": logfile.rstrip("\r\n"),
+                }
+
+        return results
+
+    def get_next_sample(self, version, build, platform, topology, test):
+        """
+        Getting the the number of the next sample for particular test results.
+        if there are no results in the DB, it will return 0
+
+        Args:
+            version (str): The version number (e.g. 4.9.0)
+            build (str): The build number (e.g. RC5-180)
+            platform (str): The platform (e.g.  Bare-Metal)
+            topology (str): The topology (e.g. 3-AZ)
+            test (str): The test name (e.g. SmallFiles)
+
+        Returns:
+             int : the number of the next sample to insert to the DB
+
+        """
+        ver_id = self.get_version_id(version=version)
+        if ver_id is None:
+            return 0
+
+        build_id = self.get_build_id(version=version, build=build)
+        if build_id is None:
+            return 0
+
+        platform_id = self.get_platform_id(platform)
+        if platform_id is None:
+            return 0
+
+        topology_id = self.get_topology_id(topology)
+        if topology_id is None:
+            return 0
+
+        test_id = self.get_test_id(test)
+        if test_id is None:
+            return 0
```

Comment:
Does return 0 is acceptable ? i.e. do we want to throw any exception or something to be handled here or we are good with this value ?

Reply:
The purpose of this function is to return the next available sample.
if there is no results in the DB the next sample number (and the first one) is 0.